pydatastructs/miscellaneous_data_structures/disjoint_set.py

```diff
@@ -74,3 +77,18 @@ def union(self, key1, key2):
 
             y_root.parent = x_root
             x_root.size += y_root.size
+
+    def make_root(self, key):
+        """
+        Finds the set to which the key belongs
+        and makes it as the root of the set.
+        """
+        if self.tree.get(key, None) is None:
+            raise KeyError("Invalid key, %s"%(key))
+
+        current_root = self.find_root(key)
+        if current_root.key != key:
+            key_set = self.tree[key]
+            current_root.parent = key_set
+            key_set.size = current_root.size
```

Comment:
I think the size of `current_root` should change. Just take the following example,
```python
>>> from pydatastructs import *
>>> dst = DisjointSetForest()
>>> dst.make_set(1)
>>> dst.make_set(2)
>>> dst.make_set(3)
>>> dst.tree[1].size
1
>>> dst.tree[2].size
1
>>> dst.tree[3].size
1
>>> dst.union(2, 3)
>>> dst.tree[1].size
1
>>> dst.tree[2].size
2
>>> dst.tree[3].size
1
>>> dst.make_root(3)
>>> dst.tree[1].size
1
>>> dst.tree[2].size # should have been 1 because now 2 is pointing to 3
2
>>> dst.tree[3].size
2
```
I think this inconsistency can cause problems when root is shifted a lot of times to different elements of set. We should keep things correct wherever possible.

Reply:
> I think this inconsistency can cause problems when root is shifted a lot of times to different elements of set. We should keep things correct wherever possible.

This can be recitifed using re-rooting technique: https://wiki.algo.is/Rerooting%20technique. Actually while changing the root I didn't tried to maintain tree/subtree structure and just focused on the size of total elements belonging to that set. However using above technique we can re-root by maintaing tree/subtree structure too

Reply:
I think we should have it. We may need to track the nodes whose size will change due to change of root. It may or not be that tricky. Try to do it by just using the parent pointers.